xarray/tests/test_dataarray.py

```diff
@@ -3075,6 +3075,24 @@ def test_propagate_attrs(self, func) -> None:
         with set_options(keep_attrs=True):
             assert func(da).attrs == da.attrs
 
+    def test_fillna_extension_array_int(self) -> None:
+        srs: pd.Series = pd.Series(
+            index=np.array([1, 2, 3]), data=pd.array([pd.NA, 1, 1])
```

Comment:
we aren't ready to support these yet. Can we instead autoconvert anything with `pd.NA` to `float` and convert to the numpy dtype otherwise?

Reply:
I added a small fix here to use arrow types as the "other" extension array type to ensure some level of better coverage, which indeed revealed a small bug.  I will update the reindexing tests as well to use them.

Reply:
The two fixes:

1. https://github.com/pydata/xarray/pull/10304/files#diff-c803294f5216cbbdffa30f0b0c9f16a7e39855d4dd309c88d654bc317a78adc0R54-R64 We didn't have a `reshape` function registered so it was falling back to `numpy` which was problematic
2. The arrow dtype indexing implementation lacked handling of tuples: https://github.com/pydata/xarray/pull/10304/files#diff-c803294f5216cbbdffa30f0b0c9f16a7e39855d4dd309c88d654bc317a78adc0R124-R127

Reply:
This issue is a lot deeper than I realized: https://github.com/pandas-dev/pandas/issues/61433, actually for the indexing issue with `arrow`.

Reply:
In any case, it looks like this doesn't have an impact here, as it turns out. It still seems like we're getting away with murder a bit because it is possible to yield 0-dimensional arrow arrays from some data types and not others.  When it is possible, that gels with what xarray expects, but when it's not possible, we have problems.  We probably need to look into a solution for this.  Maybe making `__getitem__` polymorphic, but fixing this issue + adding the test cases opened up a few different cans of worms here, so I will be able to track that issue and then can work on a fix here separately.

The broken test is similarly related to unexpected behavior around `arrow` dtypes again...will look into that as well